lib/api/core/models/repositories/roleRepository.js

```diff
@@ -118,6 +118,53 @@ module.exports = function (kuzzle) {
   };
 
   /**
+   * Get from database the document that represent the role given in parameter
+   * @param {Role} role
+   * @returns {Promise}
+   */
+  RoleRepository.prototype.loadRole = function (role) {
+    var deferred = q.defer();
+
+    if (!role._id) {
+      deferred.reject(new BadRequestError('Missing role id'));
+      return deferred.promise;
+    }
+
+    if (this.roles[role._id]) {
+      deferred.resolve(this.roles[role._id]);
+    }
+    else {
+      return this.loadOneFromDatabase(role._id);
```

Comment:
better user this.load(role._id)
It will load the role from cache if exists, or from database if not in cache